table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/AstyanaxTableDAO.java

```diff
@@ -1394,9 +1470,82 @@ public void clearStashTokenRangeSnapshot(String stashId) {
         _stashTableDao.clearTokenRanges(stashId);
     }
 
+    @Timed (name = "bv.emodb.table.AstyanaxTableDAO.listMetadataChangedTables", absolute = true)
+    @Override
+    public Iterator<ChangedTable> listMetadataChangedTables(DateTime fromInclusive, DateTime toExclusive, LimitCounter limit) {
+        checkArgument(limit.remaining() > 0, "Limit must be > 0");
+        checkArgument(fromInclusive != null, "from date cannot be null");
+        checkArgument(toExclusive != null, "to date cannot be null");
+
+        LocalDate fromDate = LocalDate.of(fromInclusive.withZone(DateTimeZone.UTC).getYear(), fromInclusive.withZone(DateTimeZone.UTC).getMonthOfYear(),
+                fromInclusive.withZone(DateTimeZone.UTC).getDayOfMonth());
+        LocalDate toDate = LocalDate.of(toExclusive.withZone(DateTimeZone.UTC).getYear(), toExclusive.withZone(DateTimeZone.UTC).getMonthOfYear(),
+                toExclusive.withZone(DateTimeZone.UTC).getDayOfMonth());
+        final int numberOfDays = (int) fromDate.until(toDate, ChronoUnit.DAYS);
+        if (numberOfDays > 30) {
+            throw new IllegalArgumentException("Specified from and to date range is greater than 30 days which is not supported. Specify a smaller range.");
+        }
+
+        List<LocalDate> dates = Stream.iterate(fromDate, d -> d.plusDays(1))
+                .limit(numberOfDays)
+                .collect(Collectors.toList());
+
+        List<ChangedTable> changedTables = Lists.newArrayList();
+        for (LocalDate date : dates) {
+            String dateKey = date.toString();
+            Map<String, Object> recordMap = _backingStore.get(_systemTableMetadataChanges, dateKey, ReadConsistency.STRONG);
+            if (recordMap != null) {
+                ObjectMapper mapper = Jackson.newObjectMapper();
+                Object tableInfo = recordMap.get("tables");
```

Comment:
All this map manipulation seems unnecessary.  Is there a way to create a POJO representation for "tables" and convert to it using`JsonHelper.convert()`?  Maybe it would be simpler if the dates weren't keys.  For example, instead of representing as:

```
"tables": {
      "review:testcustomer": {
        "18:22:42.395": "DROP",
        "18:24:09.222": "DROP"
      }
   }
```

Do something like:

```
"tables": {
      "review:testcustomer": [
         {
            "time": "18:22:42.395"
            "event": "DROP",
         },
         {
            "time": "18:24:09.222"
            "event": "DROP",
         },
      ]
   }
```